### DISCUSSION TITLE
Access zoom value

### DISCUSSION THREAD

**Discussion by Couiljer** [igordanchenko/yet-another-react-lightbox#69] [Q&A] [answered]
Hi there ! 

Thanks for this great package. 
Is there a way to access and play with the current zoom value by a ref or event ? 
(I'd like to hide all controls while zoomed in). 

cheers

**Comment by igordanchenko** [comment, marked as answer]
Hi! YARL currently doesn't expose refs to control it externally. Also, Zoom plugin resets its zoom level when the lightbox switches to the next slide. Can you describe the user experience you'd like to build?

**Reply by Couiljer** [reply]
Hi @igordanchenko ! Thanks for your reply. 

I'd like to toggle the visibility of the buttons and other UI elements (zoom +/-, next, previous, exit, title, description, ...) when I change the zoom level with pinch touch or scroll wheel :  

zoom > 0 : no UI elements, only the image
zoom === 0 : UI is back

**Reply by igordanchenko** [reply]
I can definitely add the `on.zoom` callback so you could do something along these lines:

```jsx
const [zoomedIn, setZoomedIn] = React.useState(false);

<Lightbox
  on={{
    zoom: (zoomLevel) => setZoomedIn(zoomLevel !== 1)
  }}
  render={{
    buttonPrev: zoomedIn ? () => null : undefined,
    buttonNext: zoomedIn ? () => null : undefined,
  }}
/>
```

Do you think this is the only feature you need? Are you able to hide all the UI elements you want to hide?

**Reply by Couiljer** [reply]
This would be really useful !
I could then play with the CSS values or custom render components.

**Reply by igordanchenko** [reply]
This is now available in v2.3.0.

```jsx
const [index, setIndex] = useState(-1);
const [zoomedIn, setZoomedIn] = useState(false);

// ...

<Lightbox
    slides={slides}
    open={index >= 0}
    index={index}
    close={() => setIndex(-1)}
    plugins={[Fullscreen, Slideshow, Thumbnails, Captions, Zoom]}
    on={{ zoom: (level) => setZoomedIn(level !== 1) }}
    className={zoomedIn ? "zoomedIn" : undefined}
/>
```

CSS:

```css
.zoomedIn .yarl__toolbar,
.zoomedIn .yarl__navigation_prev,
.zoomedIn .yarl__navigation_next,
.zoomedIn .yarl__thumbnails_container,
.zoomedIn .yarl__slide_captions_container{
    display: none;
}
```
